### PR TITLE
Add AddThis Sharing to organisation page.

### PR DESCRIPTION
**Pull request by ethanluoyc**
To address https://www.pivotaltracker.com/story/show/129686967

This adds AddThis sharing buttons to LocalSupport's organisation page.

### CODE REVIEW

**Comment by tansaku** [comment]
thanks @ethanluoyc so is this specific to any one persons' addthis account?

And I guess this could be added throughout the entire site?

**Comment by ethanluoyc** [comment]
This is currently my accounts, I think we need to change that though.





On Fri, Sep 2, 2016 at 7:32 PM +0800, "Sam Joseph" <notifications@github.com> wrote:












thanks @ethanluoyc so is this specific to any one persons' addthis account?



And I guess this could be added throughout the entire site?

**Comment by tansaku** [comment]
@ethanluoyc yeah - I guess we would need an AV account ... however with this fork I can perhaps show the idea to the client

**Comment by ethanluoyc** [comment]
Perhaps you can show the client first see how they feel about this. Then we can register under AV. We can also extract the link out into some config so that the code looks less cluttered

**Comment by tansaku** [comment]
hi @ethanluoyc the client seemed interested in this, but I didn't show the look and feel - I'll have to re-present this Friday

**Comment by tansaku** [comment]
@ethanluoyc how are we doing here - now we have the new account details is this ready to go?

**Comment by ethanluoyc** [comment]
Yes. I should be able to submit by this week.

File: app/views/shared/_addthis_scripts.html.erb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<!-- Go to www.addthis.com/dashboard to customize your tools -->
+<% content_for :scripts do %>
+    <%= javascript_include_tag 'http://s7.addthis.com/js/300/addthis_widget.js#pubid=ra-57c940ffd384146d' %>
```
Comment (johnnymo87) [inline app/views/shared/_addthis_scripts.html.erb]: Does this do anything that we can assert against with cucumber?

Comment (tansaku) [inline app/views/shared/_addthis_scripts.html.erb]: I would imagine it did ...

**Comment by ethanluoyc** [comment]
Hmm. Not sure I am testing it correctly, but I thought that if I add @javascript at the start of a scenario it will be able to run with JS engine enabled and I will be able to find elements dynamically generated from javascript? Maybe I am missing something.

**Comment by tansaku** [comment]
yes, although there might be timing issues - there are several debugging strategies here:

1) show me the page
2) debugger
3) selenium

I can try and give you a hand with these, but I've got to run the LS client meeting and work through a backlog of premium pull requests - will try to get you some support soon.

**Comment by ethanluoyc** [comment]
I tried to use selenium to drive the cucumber tests on my laptop with the Chrome driver. It passes the tests.

**Comment by ethanluoyc** [comment]
I think it's indeed a timing problem. If I use selenium-client to drive the cucumber test and my step definition is the following:
```ruby
Then(/^I should see the AddThis sharing sidebar$/) do
  page.should have_css '.addthis-smartlayers'
  save_and_open_screenshot
end

```
image

However, the sidebar will not be shown if I swap the two statements inside `do end`

**Comment by tansaku** [comment]
great work investigating @ethanluoyc 

I think the `have_css` has some natural waiting going on.  You might want to replace 

```rb
page.should have_css '.addthis-smartlayers'
```
with

```rb
expect(page).to have_css '.addthis-smartlayers'
```

just to be up to date with the latest rspec syntax ...

**Comment by tansaku** [comment]
hi @ethanluoyc - I can also confirm same problem locally for me - also the map fails to load:



I think there might be something weirdish going on here.  Definitely merge your branch to devlop, but the map runs for me in other instances with application.yml set up to contain the right GMAPs id ...

**Comment by ethanluoyc** [comment]
Just modified the RSpec syntax. How can we circumvent the failure of the test such that it is possible for us to merge this. It is quite a safe commit since we only added some JS to the front-end. The stability of the backend won't be affected.

**Comment by tansaku** [comment]
good question @ethanluoyc - seems like this might be good to get under the microscope ... gotta go review some other PRs first - will try to get back to this

**Comment by tansaku** [comment]
@ethanluoyc we need to merge this to develop - can you do, or perhaps you can give me access to your fork?  Thanks

**Comment by ethanluoyc** [comment]
@tansaku Added you as a collaborator on my fork :D

**Comment by tansaku** [comment]
thanks @ethanluoyc - I think I just got it all working - I'll push to your branch

**Comment by tansaku** [comment]
well, I'm thinking this is good to go if you're happy with it @ethanluoyc - checking in all the req_cache files makes it look quite bloated, but I think it makes the test more stable ...

**Comment by johnnymo87** [comment]
+1 Assuming the tests are stable now

**Pull request merged**